### PR TITLE
Init command

### PR DESCRIPTION
**Pull request by Geospace**
Following https://github.com/PoCFrance/c3pm-cli/pull/4

### CODE REVIEW

**Comment by Geospace** [comment]
@TMoynat I had to make a few architectural changes and fixed some bugs. Does it look OK to you? Thanks !

File: cmd/init.go [hunk 1]
```diff
@@ -1,29 +1,122 @@
...
+	"github.com/manifoldco/promptui"
 	"github.com/spf13/cobra"
 
-	"ctpm/constants"
+	"ctpm/projectcfg"
 )
 
 var initCmd = &cobra.Command{
-	Use:   "init",
+	Use:   "init [optionnal path]",
```
Comment (gabrielcolson) [inline cmd/init.go]: typo: `optionnal`-> `optional`

File: cmd/init.go [hunk 2]
```diff
@@ -1,29 +1,122 @@
...
 	Short: "Initialize a new c3pm project",
 
 	RunE: func(cmd *cobra.Command, args []string) error {
-		fmt.Println("User wants to initialize a project. This part will be interactive.")
+		destinationFolder := "."
+
+		// When a path is passed as argument
```
Comment (gabrielcolson) [inline cmd/init.go]: I don't like comments that describe obvious code. If you find that the code is not obvious enough, consider encapsulating it.

File: cmd/init.go [hunk 3]
```diff
@@ -1,29 +1,122 @@
...
+		if len(args) >= 1 {
+			destinationFolder = args[0]
+
+			// Try to create the folder for user
+			if _, err := os.Stat(destinationFolder); !os.IsNotExist(err) {
+				// If the directory already exists, we could use it but let's
+				// just return an error for now
```
Comment (gabrielcolson) [inline cmd/init.go]: Why? The expected behavior is to use the already existing folder.

File: cmd/init.go [hunk 4]
```diff
@@ -1,29 +1,122 @@
...
-		_ = viper.WriteConfigAs(constants.ConfigurationFileName)
+			if err := os.MkdirAll(destinationFolder, 0755); err != nil {
+				return err
+			}
+		}
+
+		// If a configuration file already exist, do not continue
+		destinationFile := filepath.Join(destinationFolder, constants.ConfigurationFileName)
+		if _, err := os.Stat(destinationFile); !os.IsNotExist(err) {
+			return errors.New("a c3pm project already exists here")
```
Comment (gabrielcolson) [inline cmd/init.go]: Maybe consider moving this error in a const instead of using a "magic string" here.

File: cmd/init.go (hunk 1 above)
Comment (Geospace) [reply cmd/init.go]: 👌

File: cmd/init.go (hunk 2 above)
Comment (Geospace) [reply cmd/init.go]: 👌

File: cmd/init.go (hunk 4 above)
Comment (Geospace) [reply cmd/init.go]: 👌

File: cmd/init.go (hunk 3 above)
Comment (Geospace) [reply cmd/init.go]: 👌 What matters is the configuration file itself.

**Pull request merged**